siatest/consensus/consts.go

```diff
@@ -0,0 +1,19 @@
+package consensus
+
+import (
+	"os"
+	"testing"
+
+	"github.com/NebulousLabs/Sia/siatest"
+)
+
+// consensusTestDir creates a temporary testing directory for a consensus. This
+// should only every be called once per test. Otherwise it will delete the
+// directory again.
+func consensusTestDir(t *testing.T) string {
+	path := siatest.TestDir("consensus", t.Name())
+	if err := os.MkdirAll(path, 0777); err != nil {
+		panic(err)
```

Comment:
it's probably okay here but panics are also generally discouraged versus returning an error because a series of deffereed calls can sometimes cause deadlocks, especially if you panic after you called lock on something while using a function that you didn't realize could fail.

If a test hits a deadlock, it'll time out after 600 or 1200 seconds instead of reporting that the true error was a failed call to mkdir.

This might be okay because mkdir really shoudln't ever be failing unless something else is seriously wrong with the filesystem.

Reply:
I know. I think in this case it's nice though since it really shouldn't ever happen during a test and it makes a lot of code cleaner.